nucleus/payara-modules/notification-core/src/main/java/fish/payara/nucleus/notification/service/HipchatNotifierService.java

```diff
@@ -0,0 +1,93 @@
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package fish.payara.nucleus.notification.service;
+
+import com.google.common.eventbus.Subscribe;
+import fish.payara.nucleus.notification.configuration.HipchatNotifier;
+import fish.payara.nucleus.notification.configuration.HipchatNotifierConfiguration;
+import fish.payara.nucleus.notification.configuration.NotifierType;
+import fish.payara.nucleus.notification.domain.HipchatNotificationEvent;
+import fish.payara.nucleus.notification.domain.execoptions.HipchatNotifierConfigurationExecutionOptions;
+import org.glassfish.api.StartupRunLevel;
+import org.glassfish.api.event.EventListener;
+import org.glassfish.api.event.EventTypes;
+import org.glassfish.hk2.runlevel.RunLevel;
+import org.jvnet.hk2.annotations.Service;
+
+import javax.annotation.PostConstruct;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.text.MessageFormat;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * @author mertcaliskan
+ */
+@Service(name = "service-hipchat")
+@RunLevel(StartupRunLevel.VAL)
+public class HipchatNotifierService extends BaseNotifierService<HipchatNotificationEvent, HipchatNotifier, HipchatNotifierConfiguration> {
+
+    private final AtomicInteger threadNumber = new AtomicInteger(1);
+    private String PREFIX = "hipchat-notification-";
+
+    public void event(Event event) {
+        if (event.is(EventTypes.SERVER_READY)) {
+            register(NotifierType.HIPCHAT, HipchatNotifier.class, HipchatNotifierConfiguration.class, this);
+        }
+    }
+
+    @Override
+    @Subscribe
+    public void handleNotification(HipchatNotificationEvent event) {
+        HipchatNotifierConfigurationExecutionOptions executionOptions =
+                (HipchatNotifierConfigurationExecutionOptions) getNotifierConfigurationExecutionOptions();
+        new Thread(new HipchatNotificationRunnable(executionOptions,
+                        event.getUserMessage(),
+                        event.getMessage()), PREFIX + threadNumber.getAndIncrement())
+                .start();
```

Comment:
as for `HttpURLConnection` I cannot cache it as its Javadoc states and don't need also since underlying connection is seamlessly reused by other instances but queue is a good idea indeed.